### PR TITLE
feat(CC): add a resource to apply central network policy

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccCentralNetworkPolicyApply_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cc -v

=== RUN   TestAccCentralNetworkPolicyApply_basic
=== PAUSE TestAccCentralNetworkPolicyApply_basic
=== CONT  TestAccCentralNetworkPolicyApply_basic
--- PASS: TestAccCentralNetworkPolicyApply_basic (105.18s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cc	105.223s

```

### CODE REVIEW

File: huaweicloud/services/cc/resource_huaweicloud_cc_central_network_policy_apply.go [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving central network policy")
+	}
+
+	getCentralNetworkPolicyApplyRespBody, err := utils.FlattenResponse(getCentralNetworkPolicyApplyResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getCentralNetworkPolicyApplyRespBody = utils.PathSearch("central_network_policies[?is_applied]|[0]",
```
Comment (ShiChangkuo) [inline huaweicloud/services/cc/resource_huaweicloud_cc_central_network_policy_apply.go:227]: we should not retrieve the first policy if none policies are not applied?
```suggestion
	getCentralNetworkPolicyApplyRespBody = utils.PathSearch("central_network_policies[?is_applied]",
```

File: huaweicloud/services/cc/resource_huaweicloud_cc_central_network_policy_apply.go [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceCentralNetworkPolicyApplyDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// get the default policy id which is not associated with the er instance and the version is 1.
```
Comment (ShiChangkuo) [inline huaweicloud/services/cc/resource_huaweicloud_cc_central_network_policy_apply.go:247]: Is it always true?
how about add a new parameter `apply` in huaweicloud_cc_central_network_policy?

File: docs/resources/cc_central_network_policy_apply.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+## Example Usage
+
+```hcl
+variable "central_network_id" {}
+variable "policy_id" {}
+
+resource "huaweicloud_cc_central_network_policy_apply" "test" {
+  central_network_id  = var.central_network_id
+  policy_id           = var.policy_id
```
Comment (ShiChangkuo) [inline docs/resources/cc_central_network_policy_apply.md]: extra space
```suggestion
  central_network_id = var.central_network_id
  policy_id          = var.policy_id
```

File: huaweicloud/services/cc/resource_huaweicloud_cc_central_network_policy_apply.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/cc/resource_huaweicloud_cc_central_network_policy_apply.go:247]: yes, it's always true;
we can create many policies, but only one can be applied.  
If the `apply`  field of both policies is true, you will not know which one is applied.

File: huaweicloud/services/cc/resource_huaweicloud_cc_central_network_policy_apply.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/cc/resource_huaweicloud_cc_central_network_policy_apply.go:227]: It return nil if none policies are not applied, so it's ok

File: docs/resources/cc_central_network_policy_apply.md (hunk 3 above)
Comment (hmbig2) [reply docs/resources/cc_central_network_policy_apply.md]: done

**Comment by hmbig2** [comment]
add a TestAccPreCheckProjectID

**Pull request merged**